File: src/com/designing/singleton/java/Singleton_LH.java
package com.designing.singleton.java;

// 懒汉
public class Singleton_LH {
    private static Singleton_LH instance;
    private Singleton_LH(){}
    public static Singleton_LH getInstance(){
        if (instance == null){
            instance = new Singleton_LH();
        }
        return instance;
    }
}

File: src/com/designing/singleton/kotlin/Singleton_LH.kt
package com.designing.singleton.kotlin

// 懒汉
class Singleton_LH private constructor(){
    companion object {
        private var instance : Singleton_LH? = null
        get() {
            if (field == null){
                field = Singleton_LH()
            }
            return field
        }
        fun get() : Singleton_LH{
            return instance!!
        }
    }
}